### PR TITLE
Upgrade Jane Street packages (DEVELOP)

### PR DESCRIPTION
**Pull request by Firobe**
Companion PR for #11011 

Contains a few more trivial fixes, mostly renaming from `Of_Xable` to `Of_Xable_without_uuid`.

### CODE REVIEW

**Comment by Firobe** [comment]
Need to upgrade toolchain images again for the CI to pass.

**Comment by Firobe** [comment]
CI is green

File: src/lib/pickles_types/at_most.ml [hunk 1]
```diff
@@ -106,7 +106,7 @@ module At_most_2 = struct
       type 'a t = ('a, Nat.N2.n) at_most
 
       include
-        Binable.Of_binable1
+        Binable.Of_binable1_without_uuid
```
Comment (psteckler) [inline src/lib/pickles_types/at_most.ml:109]: This change should be reflected in the binable version linter to assure stability of the first argument.

**Review by psteckler — APPROVED** [review]
Made comment, I'll add an issue for that.

**Comment by Firobe** [comment]
> Made comment, I'll add an issue for that.

Thanks! For reference the issue is https://github.com/MinaProtocol/mina/issues/11055

**Review by mrmr1993 — APPROVED** [review]

**Review by samuelarogbonlo — APPROVED** [review]

**Pull request merged**